### ISSUE TITLE
[Detector Support]: "Detection appears to be stuck. Restarting detection."

### ISSUE THREAD

**Issue by adamburgoyne** [blakeblackshear/frigate#8193]
### Describe the problem you are having

I've tried scouring the internet for a solution to this but I've turned up empty handed so I'm hoping this post will shed some light on my issue. 

I've occasionally noticed a `Unable to read frames from ffmpeg process` on my Reolink PoE doorbell which led me to finally taking a look at the logs. I was able to fix that by using the below config rather than the Reolink one from the documentation but I also spotted this error which I can't get to the bottom of. It happens quite frequently throughout the day and seems to heal itself but it leaves a 30ish second period where detection doesn't work. It's always the same group of logs and has no set pattern.

```
2023-10-16 18:35:12.142064989  [2023-10-16 19:35:12] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
2023-10-16 18:35:12.142369332  [2023-10-16 19:35:12] root                           INFO    : Waiting for detection process to exit gracefully...
2023-10-16 18:35:42.169249085  [2023-10-16 19:35:42] root                           INFO    : Detection process didnt exit. Force killing...
2023-10-16 18:35:42.179398793  [2023-10-16 19:35:42] root                           INFO    : Detection process has exited...
2023-10-16 18:35:45.035998678  [2023-10-16 19:35:42] detector.coral                 INFO    : Starting detection process: 1146
2023-10-16 18:35:45.036004569  [2023-10-16 19:35:42] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as usb
2023-10-16 18:35:45.044756992  [2023-10-16 19:35:45] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
```

Frigate is running as a HA addon with HA itself running on a Proxmox VM on a Lenovo M700 Tiny machine. There is a USB coral attached with the original cable, confirmed to be operating on a USB 3.0 port which is passed through as the whole USB bus to the VM.  I've tried different ports, different USB-C cables, its in a well ventilated location and there's only a Zigbee radio and bluetooth adapter also attached so power shouldn't be an issue. 

The recordings seem to carry on fine throughout these detect outages so I'm guessing the stream from the camera itself and anything go2rtc is stable and the issue lies with the USB corals configuration somewhere. I believe the coral to be working but I'm not sure if I can actually tell with one camera running - it's picked up and reports speeds in the Frigate UI so I suspect it is.

Thanks for any suggestions! 

### Version

0.12.1-367D724

### Frigate config file

```yaml
mqtt:
  host: ''
  user: ''
  password: ''
  
database:
  path: /config/frigate.db
 
ui:
  use_experimental: true
  
go2rtc:
  streams:
    doorbell: 
      - "ffmpeg:http://192.168.5.195/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=admin&password=password#video=copy#audio=copy#audio=opus#raw=-avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -flags low_delay -strict experimental -analyzeduration 1000M -probesize 1000M -rw_timeout 5000000"
    doorbell_sub: 
      - "ffmpeg:http://192.168.5.195/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=admin&password=password#video=copy#raw=-avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -flags low_delay -strict experimental -analyzeduration 1000M -probesize 1000M -rw_timeout 5000000"

cameras:
  doorbell:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/doorbell
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/doorbell_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
    record:
      enabled: True
      retain:
        days: 7
      events:
        retain:
          default: 14      
    snapshots:
      enabled: true

objects:
  track:
    - person

# Enable sound
ffmpeg:
  output_args:
    record: preset-record-generic-audio-aac
  
detectors:
  coral:
    type: edgetpu
    device: usb
```

### docker-compose file or Docker CLI command

```yaml
N/a
```

### Relevant log output

```shell
2023-10-16 18:35:12.142064989  [2023-10-16 19:35:12] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
2023-10-16 18:35:12.142369332  [2023-10-16 19:35:12] root                           INFO    : Waiting for detection process to exit gracefully...
2023-10-16 18:35:42.169249085  [2023-10-16 19:35:42] root                           INFO    : Detection process didnt exit. Force killing...
2023-10-16 18:35:42.179398793  [2023-10-16 19:35:42] root                           INFO    : Detection process has exited...
2023-10-16 18:35:45.035998678  [2023-10-16 19:35:42] detector.coral                 INFO    : Starting detection process: 1146
2023-10-16 18:35:45.036004569  [2023-10-16 19:35:42] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as usb
2023-10-16 18:35:45.044756992  [2023-10-16 19:35:45] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

USB

**Comment by adamburgoyne** [comment]
To save a potential headache, it might be due to how the coral is passed through to the VM. It's passed through as a port instead of device (due to the changing ids the usb coral has). I've spotted the actual VM logs plastered with errors around the same time as the Frigate logs so I suspect the issue is that side. I don't really want to pass through the entire PCI bus as I've heard that can cause issues so I'll keep looking.

**Comment by adamburgoyne** [comment]
Closing this, further investigation points to this being a Proxmox kernel issue. It's unable to reliably pass through the USB coral (either as a device or the single port it's on). I'm going to try a LXC container but I'm not expecting much difference, followed by passing the entire USB bus, followed by giving up and using a Pi4 or just ignoring the errors as detection _mostly_ works.

**Comment by adamburgoyne** [comment]
I've closed this as the issue is caused by Proxmox and not Frigate. The logs in Frigate are due to the Proxmox host resetting the usb device being passed through, not sure why though. I couldn't find many posts that had a direct link between this log and the actual problem though so hopefully leaving this post here with the below solution will help others who also first spot the error via Frigate logs. 

Since posting this I've managed to setup a LXC container thanks to the below linked youtube video which was much easier than I was expecting, about 10 minutes work with my limited proxmox knowledge. So far so good, no USB crashes, a clear frigate log, significantly lower inference speed (9ms vs 30ms in the HA VM) and slightly lower CPU usage. Time will tell but I'm pretty happy with this solution so far and no need to pass through the entire USB controller (with associated issues).

https://www.youtube.com/watch?v=DmbFq5dMsFo
